src/NLog/Targets/EventLogTarget.cs

```diff
@@ -349,7 +349,9 @@ internal string GetFixedSource()
         /// <returns></returns>
         private EventLog GetEventLog(LogEventInfo logEvent)
         {
-            return eventLogInstance ?? (eventLogInstance = new EventLog(this.Log, this.MachineName, this.Source.Render(logEvent)));
+            if (eventLogInstance?.Source == this.Source.Render(logEvent))
```

Comment:
In worst case we're rendering the `source` two times?

Also there is an (implicit) two times check on `null` and we need to check also log and machine?